pandas/io/parquet.py

```diff
@@ -92,7 +92,7 @@ def write(
         **kwargs,
     ):
         self.validate_dataframe(df)
-        path, _, _, _ = get_filepath_or_buffer(path, mode="wb")
+        path, _, _, should_close = get_filepath_or_buffer(path, mode="wb")
```

Comment:
For some reason we only have this logic in read. Should add to write too

Reply:
hmm, can you open an issue for this

Reply:
For which bit? File basically needs to be closed post write for s3fs to throw as expected. @TomAugspurger mention it here: https://github.com/pandas-dev/pandas/issues/32486#issuecomment-595992013

Reply:
> For some reason we only have this logic in read. Should add to write too

your comment here.